CMakeLists.txt

```diff
@@ -8,13 +8,20 @@ set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
 add_library(resilience)
 add_library(Kokkos::resilience ALIAS resilience)
 
-# Require C++14
-if (NOT CMAKE_CXX_COMPILER MATCHES ".*nvcc_wrapper" )
-  target_compile_features(resilience PUBLIC cxx_std_14)
+
+option(KR_ALL_WARNINGS "Enable all warnings" ON)
+option(KR_WARNINGS_AS_ERRORS "Enable warnings as errors" ON)
```

Comment:
Do you really want an option for these?  `-DCMAKE_CXX_FLAGS="-Wall -Werror"` is easy enough IMO.

Reply:
Also has nothing to do with the bumping Kokkos to 4.0

Reply:
That's true, but it's a good way to avoid compiler differences. CMake even added a property in more recent versions for Werror. As far as Kokkos 4.0 updates sadly you are correct. It's just we had... a lot of problems with the current way the CMake was set up and not making warnings not errors by default (previous behavior) solves them